pymc/util.py

```diff
@@ -16,21 +16,59 @@
 import re
 
 from collections import namedtuple
-from collections.abc import Sequence
+from collections.abc import Hashable, Mapping, Sequence
 from copy import deepcopy
-from typing import cast
+from types import EllipsisType
+from typing import TypeAlias, cast
 
 import arviz
 import cloudpickle
 import numpy as np
 import xarray
 
 from cachetools import LRUCache, cachedmethod
-from pytensor import Variable
 from pytensor.compile import SharedVariable
+from pytensor.graph.basic import Variable
+from pytensor.tensor.variable import TensorVariable
 
 from pymc.exceptions import BlockModelAccessError
 
+CoordValue: TypeAlias = Sequence[Hashable] | np.ndarray | None
+# User-provided values for a single coordinate dimension.
```

Comment:
Also on the others, this is a docstring for a variable:
```suggestion
"""User-provided values for a single coordinate dimension."""
```

Reply:
done the changes!!! @michaelosthege